include/http_client.h

```diff
@@ -0,0 +1,22 @@
+#ifndef PROXYSQL_HTTP_CLIENT_H
+#define PROXYSQL_HTTP_CLIENT_H
+
+#include <mongoose.h>
```

Comment:
Added doc explaining the use of mongoose and how http_client.h adds value to that.